### DISCUSSION TITLE
Add field layouts to link fields

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#15541] [Ideas] [answered]
Current link field which allows linking to custom urls or specific craft element is big improvement compared to previous url field.

However in my opinion it is still not ready to replace plugins such as Hyper in many cases. That's because it does not allow setting custom link text and target attribute which are very common usages of  such link fields in my experience.

Of course we can use additional fields to set these parameters, but if we need to create separate fields, why use link field at all and not some custom matrix instead?

That's why i thin it would be very useful if we could add custom fields to link field, with some built-in ones (just like address field has some built in attributes), such as target and link text.

**Comment by brandonkelly** [comment, marked as answer]
Link is not intended to be a Hyper replacement. It does one job only: store the link itself.

If we wanted to add custom field support, it would require that links be elements, at which point it’s adding just as much complexity (and performance load) as a Matrix field.
